### ISSUE TITLE
Detect Android properly for Samsung mobile browser

### ISSUE THREAD

**Issue by markschlosseratbentley** [iTwin/itwinjs-core#3712]
I have attached the userAgent printouts to this defect.

The isAndroidBrowser function in ProcessDetector fails to return true on the Samsung Browser.  The current code is:
`
  public static get isAndroidBrowser() { return this.isBrowserProcess && /android/i.test(window.navigator.userAgent); }`

So it is basically checking for android in the userAgent.  See the attached the info I see on the Samsung tablet with both browsers.  The only difference is in the User Agent line, where “Android 11; SM-T870” is missing in Samsung’s browser, but X11 is added, and later SamsungBrowser/14.2.

**Comment by tcobbs-bentley** [comment]
@markschlosseratbentley I don't think that this is something we want to fix. It's fairly straightforward to search for "samsungbrowser" in the user agent, but the browser is also [used](https://developer.samsung.com/internet/user-agent-string-format.html) in their Tizen smart TV OS. We could then say something like "includes samsungbrowser and does not include tizen", but IMO that just assumes they won't arbitrarily release it on some other platform. Furthermore, the official Samsung page linked states that their user agent is supposed to include "Android" in the OS part.

UserAgents.io also seems to [inicate](https://useragents.io/uas/mozilla-5-0-linux-android-11-samsung-sm-g973u-applewebkit-537-36-khtml-like-gecko-samsungbrowser-14-2-chrome-87-0-4280-141-mobile-safari-537-36_89d3f6f52a46a632cd42a30a63f8b73b) that "Android" is usually present in the string, although they do have [some](https://useragents.io/uas/mozilla-5-0-x11-linux-x86-64-applewebkit-537-36-khtml-like-gecko-samsungbrowser-18-0-chrome-99-0-4844-88-safari-537-36_9bf7f0721133eb21266a092fc49e388a) [X11](https://useragents.io/uas/mozilla-5-0-x11-linux-x86-64-applewebkit-537-36-khtml-like-gecko-samsungbrowser-26-0-chrome-122-0-0-0-safari-537-36_6e7e430ff4ae61e18e82f059cfa0c8f6) [entries](https://useragents.io/uas/mozilla-5-0-x11-linux-x86-64-applewebkit-537-36-khtml-like-gecko-samsungbrowser-20-0-chrome-106-0-5249-126-safari-537-36_c5541153dd3b4c4a2424b7f6ca1b6529). Note: Google only has 3 hits on useragents.io that include "x11" in their titles and "samsung browser" in their text. There are at least 9 pages of hits with "android" in the title instead of "x11".

Based on another Google search, it [appears](https://android.stackexchange.com/questions/258313/why-is-linux-device-showing-on-googles-your-devices-when-using-samsung-browse) that the user agent you are seeing is due to the "Desktop web site" option being enabled on the device, so I would argue that iTwin is behaving correctly, since Samsung intentionally hides the fact that it is running on Android in this case.

If you have Android-only rendering code, is it possible to instead rely on the information returned by WebGL, instead of relying on this? I truly believe that iTwin is currently behaving correctly.

**Comment by markschlosseratbentley** [comment]
I am okay with this being left as-is. If we run into any major issues reported with the Samsung Browser, we can look into relying on some WebGL information as you suggest.
